goon/browserassets/js/browserOutput.js

```diff
@@ -171,25 +171,67 @@ function setHighlightColor(match) {
 	match.style.background = opts.highlightColor
 }
 
+function escapeRegex(input){ // put this in a function incase it ever needs to be used elsewhere, makes code cleaner
+	return input.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
 //Highlights words based on user settings
 function highlightTerms(el) {
 	if(regexHasError) return; //just stop right there ig the regex is gonna except
+	function highlightRecursor(element,term){ //recursor function to do the highlighting proper
+		var regex = new RegExp("("+term+")","gi")
+		function replace(str) {
+			return str.replace(regex,'<span style="background-color:'+opts.highlightColor+'">$&</span>')
+		}
+		var s=''
+		var work=element.innerHTML;
+		var work_lc=work.toLowerCase()
+		var ind=0;
+		while(ind < work_lc.length) {
+			console.log(s);
+			var next_term=work_lc.substring(ind).search(regex);
+            if(next_term!=-1)next_term+=ind;
+			var next_tag =work_lc.indexOf('<',ind)
+			if(next_tag == -1) {
+				s+=replace(work.substring(ind));
+				break;
+			}
+			else if(next_term==-1) {
+				s+=work.substring(ind)
+				break;
+			}
+			else if(next_tag < next_term) {
+				var temp=work_lc.indexOf('>',next_tag)
```

Comment:
The html of `one <span foo="bar>baz"> two </span> tree` is technically correct and well-formed, yet your algorithm would mangle it when trying to highlight `baz`.
I *think* we're not using this kind of thing anywhere, but I can't be certain...

As I've mentioned several times already the correct way to handle this to navigate the DOM tree using js-provided capabilities, and not via regex...

But this would probably work in the short term, yes...

Reply:
stray braces will NEVER appear, ever unless there's a security vulnerability, or an admin is being an idiot, theyre all filtered out already at the server side

Reply:
I actually tested this and I haven't been able to cause this cause a problem.